ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.tsx

```diff
@@ -12,16 +15,75 @@ export type DialogConfig = {
 })
 export class OntotextDialogWebComponent {
 
+  private documentOverflow: string
+  private closeButton: HTMLButtonElement;
+
+  @Element() hostElement: HTMLElement;
+
   @Prop() config: DialogConfig;
 
+  /**
+   * Handles the Escape and Tab keys:
+   * <ol>
+   *   <li> Escape - Calls the close action</li>
+   *   <li> Tab - Controls the focused elements to be only the dialog elements.</li>
+   *
+   * @param ev The keyboard event.
+   */
+  @Listen('keydown', {target: "document"})
+  keydownListener(ev: KeyboardEvent) {
+    if (ev.key === 'Escape') {
+      this.closeButton.click();
+    }
+
+    // Handle Tab key press to manage focus within the dialog
+    if (ev.key === 'Tab') {
```

Comment:
This approach is far more complicated that it should be IMO. Can't we just constrain the focus in the dialog preventing it from popping-out to elements outside? https://stackoverflow.com/a/51210194 check this approach

Reply:
Done